frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.tsx

```diff
@@ -28,39 +28,17 @@ function BreakoutStep({
 }: NotebookStepUiComponentProps) {
   const { stageIndex } = step;
 
-  const clauses = Lib.breakouts(topLevelQuery, stageIndex);
+  const breakouts = Lib.breakouts(topLevelQuery, stageIndex);
 
-  const checkColumnSelected = (
-    columnInfo: Lib.ColumnDisplayInfo,
-    breakoutIndex?: number,
-  ) => {
-    return (
-      typeof breakoutIndex === "number" &&
-      columnInfo.breakoutPosition === breakoutIndex
-    );
-  };
-
-  const getColumnGroups = (breakoutIndex?: number) => {
```

Comment:
Heavy calculations, moved to the popover to make sure they run already when it's opened